diff --git a/src/server/ua_services_view.c b/src/server/ua_services_view.c
index 3a5e861..de4d770 100644
--- a/src/server/ua_services_view.c
+++ b/src/server/ua_services_view.c
@@ -96,6 +96,7 @@ Service_Browse_single(UA_Server *server, UA_Session *session,
     size_t referencesIndex = 0;
     /* set the browsedescription if a cp is given */
     UA_UInt32 continuationIndex = 0;
+    struct ContinuationPointEntry *cpLoop = NULL, *cpLast = NULL;
     if(cp) {
         descr = &cp->browseDescription;
         maxrefs = cp->maxReferences;
@@ -215,6 +216,18 @@ Service_Browse_single(UA_Server *server, UA_Session *session,
             UA_ByteString_copy(&cp->identifier, &result->continuationPoint);
         }
     } else if(maxrefs != 0 && referencesCount >= maxrefs) {
+        if (session->availableContinuationPoints <= 0) {
+            // if no more ContinuationPoints are available,
+            // we delete the last one
+            LIST_FOREACH(cpLoop, &session->continuationPoints, pointers) {
+                cpLast = cpLoop;
+            }
+
+            if (cpLast) {
+                removeCp(cpLast, session);
+            }
+        }
+
         /* create a cp */
         if(session->availableContinuationPoints <= 0 ||
            !(cp = UA_malloc(sizeof(struct ContinuationPointEntry)))) {
