diff --git a/src/server/ua_nodestore.h b/src/server/ua_nodestore.h
index 6008432..6d431b9 100644
--- a/src/server/ua_nodestore.h
+++ b/src/server/ua_nodestore.h
@@ -41,8 +41,6 @@ void UA_NodeStore_delete(UA_NodeStore *ns);
 UA_Node * UA_NodeStore_newNode(UA_NodeClass nodeClass);
 #define UA_NodeStore_newObjectNode() \
     (UA_ObjectNode*)UA_NodeStore_newNode(UA_NODECLASS_OBJECT)
-#define UA_NodeStore_newVariableNode() \
-    (UA_VariableNode*)UA_NodeStore_newNode(UA_NODECLASS_VARIABLE)
 #define UA_NodeStore_newMethodNode() \
     (UA_MethodNode*)UA_NodeStore_newNode(UA_NODECLASS_METHOD)
 #define UA_NodeStore_newObjectTypeNode() \
@@ -56,6 +54,17 @@ UA_Node * UA_NodeStore_newNode(UA_NodeClass nodeClass);
 #define UA_NodeStore_newViewNode() \
     (UA_ViewNode*)UA_NodeStore_newNode(UA_NODECLASS_VIEW)
 
+/* Enable read/write AccessLevel flags for variables by default. More complete
+ * access control is added starting in the 0.3 branch. */
+static UA_VariableNode *
+UA_NodeStore_newVariableNode(void) {
+    UA_VariableNode *vn = (UA_VariableNode*)
+        UA_NodeStore_newNode(UA_NODECLASS_VARIABLE);
+    if(vn)
+        vn->accessLevel = 3; /* Enable read/write */
+    return vn;
+}
+
 /* Delete an editable node. */
 void UA_NodeStore_deleteNode(UA_Node *node);
 
