diff --git a/plugins/ua_network_tcp.c b/plugins/ua_network_tcp.c
index 6f27020..96e3329 100644
--- a/plugins/ua_network_tcp.c
+++ b/plugins/ua_network_tcp.c
@@ -395,8 +395,8 @@ ServerNetworkLayerTCP_start(UA_ServerNetworkLayer *nl, UA_Logger logger) {
                                         "opc.tcp://%s:%d", hostname, layer->port);
 #endif
         du.data = (UA_Byte*)discoveryUrl;
+      UA_String_copy(&du, &nl->discoveryUrl);
     }
-    UA_String_copy(&du, &nl->discoveryUrl);
 
     /* Create the server socket */
     SOCKET newsock = socket(PF_INET, SOCK_STREAM, 0);
