diff --git a/src/ua_securechannel.c b/src/ua_securechannel.c
index 1da7b6a..7ed56ea 100644
--- a/src/ua_securechannel.c
+++ b/src/ua_securechannel.c
@@ -196,6 +196,9 @@ UA_SecureChannel_sendChunk(UA_ChunkInfo *ci, UA_ByteString *dst, size_t offset)
 UA_StatusCode
 UA_SecureChannel_sendBinaryMessage(UA_SecureChannel *channel, UA_UInt32 requestId,
                                    const void *content, const UA_DataType *contentType) {
+    if(!channel)
+        return UA_STATUSCODE_BADINTERNALERROR;
+
     UA_Connection *connection = channel->connection;
     if(!connection)
         return UA_STATUSCODE_BADINTERNALERROR;
