diff --git a/plugins/historydata/ua_history_data_backend_memory.c b/plugins/historydata/ua_history_data_backend_memory.c
index dbd6d4169..50a090224 100644
--- a/plugins/historydata/ua_history_data_backend_memory.c
+++ b/plugins/historydata/ua_history_data_backend_memory.c
@@ -208,9 +208,17 @@ serverSetHistoryData_backend_memory(UA_Server *server,
     }
     UA_DateTime timestamp = 0;
     if (value->hasSourceTimestamp) {
-        timestamp = value->sourceTimestamp;
+        if (value->sourceTimestamp == 116444736000000000){
+            return UA_STATUSCODE_GOOD;
+        } else {
+            timestamp = value->sourceTimestamp;
+        }
     } else if (value->hasServerTimestamp) {
-        timestamp = value->serverTimestamp;
+        if (value->serverTimestamp == 116444736000000000){
+            return UA_STATUSCODE_GOOD;
+        } else {
+            timestamp = value->sourceTimestamp;
+        }
     } else {
         timestamp = UA_DateTime_now();
     }
